code/graph-algorithms/diameter_trees/cpp/Diameter.cpp

```diff
@@ -0,0 +1,65 @@
+//Learn O(n)
+
+
+// The following is O(n^2)
+#include<bits/stdc++.h>
```

Comment:
It is not preferable to use this header as it is not supported by all compilers (#1915).